Dockerfile

```
@@ -1,21 +1,49 @@
-FROM python:3.4
+FROM ubuntu:18.04 as builder
 
-RUN pip install pipenv==8.2.7 \
-  && pip install awscli==1.11.174
-RUN apt update && apt install -y libsnappy-dev
+ARG APP_VERSION=unknown
+
+RUN apt-get update \
+    && apt-get install -y curl git locales make build-essential
+
+RUN locale-gen en_US.UTF-8
+
+ENV LANG=en_US.UTF-8
+RUN curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add - \
+    && echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list \
+    && curl -sL https://deb.nodesource.com/setup_8.x | bash -
+
+RUN apt-get update \
+    && apt-get install -y yarn nodejs
 
-RUN mkdir -p /usr/src/app
 WORKDIR /usr/src/app
 
-ENV AWS_DEFAULT_REGION eu-west-1
+COPY . /usr/src/app
+RUN yarn compile
+RUN ./scripts/translate_schemas.sh
+RUN printf $APP_VERSION > .application-version
 
-COPY Pipfile Pipfile
-COPY Pipfile.lock Pipfile.lock
+# We don't want node_modules to be copied to the runtime image
+RUN rm -rf node_modules
+###############################################################################
+# Second Stage
+###############################################################################
 
-RUN pipenv install --deploy --system
+FROM python:3.4-slim-stretch
 
 EXPOSE 5000
 
-ENTRYPOINT ["sh", "docker-entrypoint.sh"]
+RUN apt update && apt install -y libsnappy-dev build-essential
 
-COPY . /usr/src/app
+RUN mkdir -p /usr/src/app
+WORKDIR /usr/src/app
+
+COPY Pipfile.lock Pipfile.lock
+COPY .python-version .python-version
+
+RUN pip install pipenv==8.2.7
```

About time we changed this?
```suggestion
RUN pip install pipenv==2018.11.26
```

---

Nope